test/lit.cfg

```diff
@@ -583,6 +583,10 @@ elif platform.system() == 'Linux':
     if swift_test_mode != 'only_non_executable':
       config.available_features.add('swift_interpreter')
 
+# Check if running on Android by seeing if the ANDROID_DATA environment variable
+# is set.
+run_android = 'ANDROID_DATA' in os.environ
```

Comment:
Can you check if Python’s `platform.dist()` might give you this information and we avoid the “random” environment variable?

Reply:
[platform.dist is deprecated](https://docs.python.org/2/library/platform.html#platform.dist): [this SO answer showed people checking env variables instead](https://stackoverflow.com/a/48020139), so I did the same. It's understandable given the Python devs don't expect it to run on Android.

Reply:
Yes, I read that it was deprecated in 3.5 and going to be removed in 3.8, but right now Swift needs 2.7 to build, so it is not going to affect us (I'm trying to be compatible with 3.5, but we are still far away from that goal, or being able to stop using deprecated methods).

Reply:
Nope, read my two links: deprecated since 2.6 also and didn't give the needed info anyway.

I was happy to see your pull to add support for Python 3 as that was a PITA when configuring the Swift build on linux and Android and currently undocumented, which is why I explicitly called it out above.

Reply:
Sorry, the documentation for the [2.x series](https://docs.python.org/2.7/library/platform.html#unix-platforms) and the [3.x series](https://docs.python.org/3/library/platform.html#unix-platforms) differ, and I was looking at the later.

Use `linux_distribution` instead. `dist` was deprecated by renaming to `linux_distribution`. They are the same function. It is still deprecated in 3.5, but I think we can live with it for a little while.

Reply:
linux_distribution doesn't work on Android either, as [it looks for](https://github.com/python/cpython/blob/2.7/Lib/platform.py#L356) an /etc [release file](https://github.com/python/cpython/blob/2.7/Lib/platform.py#L278), which Android doesn't have.

ANDROID_DATA in the environment appears to be the best bet, as [it's set by the OS itself](https://android.googlesource.com/platform/system/core/+/master/rootdir/init.environ.rc.in).

Reply:
Done.